### PR TITLE
Data flowing thru fullstack into all components based on state/product_id

### PR DESCRIPTION
**Pull request by LachelleF**
Modified schema, server and expanded components work to support state and responsiveness to product_id. Will focus on UI next as well as pulling product_id from URL per Rane's guidance next.

### CODE REVIEW

File: client/components/App.jsx [hunk 1]
```diff
@@ -1,18 +1,85 @@
...
-    this.state = {};
+    this.state = null;
+  }
+
+  fetchImage(productId) {
+    return "url1";
+  }
+
+  fetchProductPlatform(productId) {
+    return { os: ["urlLinux", "urlWindows"] };
```
Comment (Chris-Hesterman) [inline client/components/App.jsx:26]: there is the endpoint for platforms, but os is not included in the results, Platforms are like Steam, Origin etc.  Os is available from main endpoint, unless you would like for there to be os included in the platforms endpoint.  I might just do that anyway shouldn't be difficult.  Let me know

File: client/components/Platforms.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const PlatformsStyled = styled.div`
+  width: 338.91px;
+  height: 22.31px;
+`;
```
Comment (Chris-Hesterman) [inline client/components/Platforms.jsx:7]: the images from platforms are in png format, and they are all 24px x 24px .  That can be tweeked as necessary either on your end or mine.
the 24px x 24px includes some padding around the actual 'icon' shape.

File: client/components/App.jsx (hunk 1 above)
Comment (LachelleF) [reply client/components/App.jsx:26]: omg, would you? That would be so great, Chris. If you're sure it's not too much trouble.

File: client/components/Platforms.jsx (hunk 2 above)
Comment (LachelleF) [reply client/components/Platforms.jsx:7]: OK ty for the guidance ... and the padding :).

**Pull request merged**